### ISSUE TITLE
Increase Stack Size With DOSBox-X

### ISSUE THREAD

**Issue by GusCE6** [joncampbell123/dosbox-x#3721]
### Question

Sirs,

Recently I learned how to compile the source code for AGT (Adventure Gaming Tool), an all-text game engine.

Compiling the PA4 files for the RUNB.EXE file worked like a charm, but whenever I try to compile for COMPILEB.EXE I always get an ERROR 71: INTERNAL STACK OVERFLOW error message in the COMMAND.PA3 file. I have tried memory adjustments, changing the {$Whatever-/+} switches in the files, etc., but nothing works. The only thing left is to increase the STACK size, which DOSBox does not have. When I saw CONFIG.SYS in DOSBox-X I was hopeful but- nothing there.

AGT uses Turbo PASCAL. The compiler I'm using is "Turbo PASCAL (with DOSBox)," v 7.3.5 by Luu Nguyen Thien Hau. It appears to be the latest version from 2020.

I have adapted the files so DOSBox-X is used instead, but without that STACK size increase option it won't work any better.

So far I've dealt with it by removing some of the stack. But I want it all in there.

This is a very basic feature so I'm assuming I'm missing something here. Can anyone help? Thanks in advance. 

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by Jookia** [comment]
DOS doesn't manage the stack does it? The application does.

**Comment by maron2000** [comment]
I ran Turbo Pascal on PC-DOS2000 with `STACKS=64,512` but no luck, so I think `STACKS` doesn't matter.
Simply compile using TPC.EXE. 
![command_000](https://user-images.githubusercontent.com/68574602/189328319-e5a58d2f-e560-4897-9cf5-579727cc09e3.png)

**Comment by Jookia** [comment]
Does this happen on FreeDOS?

**Comment by maron2000** [comment]
I couldn't get to run FreeDOS with `STACKS` value changed, but I got `ERROR 71: INTERNAL STACK OVERFLOW` on default settings.
I ran FreeDOS 1.3 RC5 on VMware to test this.
You can compile the source using `tpc.exe` on FreeDOS as well.

Are we hearing anything from @GusCE6 ?

**Comment by Jookia** [comment]
So is this a dosbox-x bug?

**Comment by maron2000** [comment]
@Jookia 
No, I think its better to understand that this is a bug of Turbo Pascal (TPX.EXE).
As I mentioned, compiling on PC-DOS and FreeDOS shows the same error as doing so on DOSBox-X, that means the emulation is accurate.
Maybe compiling using another version of Turbo Pascal might work, if you want to use TPX.EXE.

**Comment by GusCE6** [comment]
Hello.

I am sorry for the lengthy delay, but illness, a relative holding up important legal matters, and the fact I lost my password and was unable to reset it with the browsers I was using on Puppy Linux caused this delay.

I am curious about this "TPX.EXE." Where can I download it to try? I would really like to get everything compiled, including my additions.

Again, thank you- this website is confusing to use...

**Comment by maron2000** [comment]
@GusCE6 TPX.exe is included in Turbo Pascal, but as I mentioned above, I haven't found a way to compile AGT without errors.
I succeeded in compiling it using TPC.exe (also included in Turbo Pascal), as I showed above with a screenshot.

**Comment by GusCE6** [comment]
All right, where to download the version of TPE.EXE you used, or the entire package? It must be usable by DOSBox or DOSBox-X.

Thank you.

**Comment by emendelson** [comment]
I don't think this site publishes links to copyrighted software. You can easily search the web, of course.

**Comment by GusCE6** [comment]
I'm not quite sure I understand this.

Is "TPE.EXE" a lone file? So far I cannot find any such thing, and I've looked.

Do you just drag it to DOSBox or DOSBox-X and take it from there?

What kind of a program is TPE.EXE?
